src/Common/Core/Impl/OS/Win32EnvironmentBlock.cs

```diff
@@ -0,0 +1,102 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Text;
+using static Microsoft.Common.Core.NativeMethods;
+
+namespace Microsoft.Common.Core.OS {
+    public class Win32EnvironmentBlock : Dictionary<string, string> {
```

Comment:
Ok, but for that you don't need to implement `IDictionary` - you need a pair of `get`/`set` methods or an indexer. The fact that you use a dictionary to store key/value pairs (or maybe `ConcurrentDictionary`) isn't important for the code that uses `Win32EnvironmentBlock` and can be encapsulated.